src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -49,16 +49,26 @@ void ExtractUriHostPath(const std::string& uri, std::string* host,
 
 }  // namespace
 
-JwtAuthenticator::JwtAuthenticator(Upstream::ClusterManager& cm,
-                                   JwtAuthStore& store)
+JwtAuthenticator::JwtAuthenticator(Upstream::ClusterManager &cm,
+                                   JwtAuthStore &store)
     : cm_(cm), store_(store) {}
 
 // Verify a JWT token.
-void JwtAuthenticator::Verify(HeaderMap& headers,
-                              JwtAuthenticator::Callbacks* callback) {
+void JwtAuthenticator::Verify(HeaderMap &headers,
+                              JwtAuthenticator::Callbacks *callback) {
   headers_ = &headers;
   callback_ = callback;
 
+  // Per the spec
+  // http://www.w3.org/TR/cors/#cross-origin-request-with-preflight-0, CORS
+  // pre-flight requests shouldn't include user credentials.
+  if (headers_->Method() && Http::Headers::get().MethodValues.Options ==
```

Comment:
This bypasses too many requests, e.g. WebDAV uses OPTIONS for different purpose and those shouldn't be bypassed.

Reply:
https://github.com/istio/proxy/pull/2140 is created to add more checks for CORS requests.